apps/audit/middleware.py

```diff
@@ -0,0 +1,24 @@
+import uuid
+
+from django.conf import settings
+
+from apps.audit.transaction import audit_transaction
+
+
+class AuditTransactionMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        transaction_id = get_audit_transaction_id(request)
+        with audit_transaction(transaction_id):
+            return self.get_response(request)
+
+
+def get_audit_transaction_id(request):
```

Comment:
Just to make sure I understand correctly, we don't want to have this transaction id to live longer than a single request, since we want to be able to tell which models were deleted together and by which request?

Reply:
If the headers exist then they may be present in other logs e.g. load balancers / WAF and so it's useful to be able to tie them together.